tests/Microsoft.Data.DataFrame.Tests/DataFrameTests.cs

```diff
@@ -598,5 +598,66 @@ public void TestPrimitiveColumnSort()
             Assert.Null(sortedIntColumn[9]);
 
         }
+
+        [Fact]
+        public void TestJoin()
+        {
+            DataFrame left = MakeDataFrameWithAllColumnTypes(10);
+            DataFrame right = MakeDataFrameWithAllColumnTypes(5);
+
+            // Tests with right.RowCount < left.RowCount
+            // Left join
+            DataFrame join = left.Join(right);
+            Assert.Equal(join.RowCount, left.RowCount);
+            Assert.Equal(join.ColumnCount, left.ColumnCount + right.ColumnCount);
+            Assert.Null(join["Int_right"][6]);
```

Comment:
I would have expected better validation here than just checking a single value. Shouldn't we be validating that the whole DataFrame is correct?

Reply:
Done.